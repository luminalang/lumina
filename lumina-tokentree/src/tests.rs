use super::*;
use insta;

macro_rules! tree {
    ($name:ident, $src:literal) => {
        #[test]
        fn $name() {
            lumina_util::test_logger();
            let mut parser = Parser::new($src);
            let entity = parser.next(false);
            insta::assert_snapshot!(entity);
        }
    };
}

tree!(precedence, "1 = 2, 3 . 4 + 4 + 5 * 5, 6 + 6");
tree!(
    nested_match,
    "match 0
| outer0 -> r0
| outer1 ->
  match 1
  | inner0 -> 10
  | inner1 -> 11
| outer2 ->
  match 2
  | inner0 -> 20
  | inner1 -> 21
"
);
