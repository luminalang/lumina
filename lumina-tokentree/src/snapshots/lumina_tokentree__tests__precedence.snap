---
source: lumina-tokentree/src/tests.rs
expression: entity
snapshot_kind: text
---
[ 1
, Header(=)
    Operators(2)
      2
    Op(0, ,)
      Operators(1)
        3
      Op(0, .)
        Operators(1)
          Operators(2)
            4
          Op(0, +)
            4
          Op(1, +)
            5
        Op(0, *)
          5
    Op(1, ,)
      Operators(1)
        6
      Op(0, +)
        6
]
